python/ray/_private/arrow_utils.py

```diff
@@ -1,13 +1,56 @@
 import json
+import logging
+import os
 from typing import Dict, Optional
 from urllib.parse import parse_qsl, unquote, urlencode, urlparse, urlunparse
 
 from packaging.version import Version, parse as parse_version
 
+_RAY_DISABLE_PYARROW_VERSION_CHECK = "RAY_DISABLE_PYARROW_VERSION_CHECK"
+
+
 _PYARROW_INSTALLED: Optional[bool] = None
 _PYARROW_VERSION: Optional[Version] = None
 
 
+# NOTE: Make sure that these lower and upper bounds stay in sync with version
+# constraints given in python/setup.py.
+# Inclusive minimum pyarrow version.
+_PYARROW_SUPPORTED_VERSION_MIN = "9.0.0"
+_PYARROW_VERSION_VALIDATED = False
+
+
+logger = logging.getLogger(__name__)
+
+
+def _check_pyarrow_version():
```

Comment:
Does it need to be inside core? since it's only used by library.

Reply:
then we should put in the `_common` folder.

Reply:
I think that makes sense, but i think we'd move both `get_pyarrow_version` and `_check_pyarrow_version` together (not sure it makes sense to split them up